contracts/terraswap_router/src/querier.rs

```diff
@@ -19,3 +21,25 @@ pub fn compute_tax(querier: &QuerierWrapper, amount: Uint128, denom: String) ->
         tax_cap,
     ))
 }
+
+pub fn compute_reverse_tax(
+    querier: &QuerierWrapper,
+    amount: Uint128,
+    denom: String,
+) -> StdResult<Uint128> {
+    if denom == "uluna" {
+        return Ok(Uint128::zero());
+    }
+
+    let terra_querier = TerraQuerier::new(querier);
+    let tax_rate: Decimal = (terra_querier.query_tax_rate()?).rate;
+    let tax_cap: Uint128 = (terra_querier.query_tax_cap(denom)?).cap;
+
+    let tax: Uint128 = (std::cmp::min(
+        Uint256::from(amount).div(Decimal256::one() - Decimal256::from(tax_rate)),
```

Comment:
When we compute send_amount from total_amount, we use
```
total_amount = send_amount * tax_rate + send_amount = send_amount * (1 + tax_rate)
```

but current equation seems using
```
total_amount = send_amount / (1 - tax_rate)
```

Reply:
so I think, it should be
```rust
Uint256::from(amount).mul(Decimal256::one() + Decimal256::from(tax_rate))
```

Reply:
like here,

https://github.com/terraswap/terraswap/blob/97cefa337798bdb0cba0327dd4152607839a5c77/packages/terraswap/src/asset.rs#L42

Reply:
Ah, I was confused with send_amount. fixed it. https://github.com/terraswap/terraswap/pull/30/commits/36e90e3e7b3ee37e9f892dc37cb7d2d9bfb304b8